document-generator-prosecution/src/test/java/uk/gov/companieshouse/document/generator/prosecution/handler/ProsecutionHandlerTest.java

```diff
@@ -0,0 +1,73 @@
+package uk.gov.companieshouse.document.generator.prosecution.handler;
+
+import com.sun.org.apache.regexp.internal.RE;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.api.model.prosecution.prosecutioncase.ProsecutionCaseStatusApi;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.document.generator.prosecution.ProsecutionType;
+import uk.gov.companieshouse.document.generator.prosecution.exception.HandlerException;
+import uk.gov.companieshouse.document.generator.prosecution.exception.ProsecutionServiceException;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.ProsecutionDocument;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.defendant.Defendant;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.offence.Offence;
+import uk.gov.companieshouse.document.generator.prosecution.mapping.model.prosecutioncase.ProsecutionCase;
+import uk.gov.companieshouse.document.generator.prosecution.service.ProsecutionService;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ProsecutionHandlerTest {
+
+    private static final String RESOURCE_URI = "/internal/company/1234/prosecution-cases/4321/defendants/1122";
+    private static final String REQUEST_ID = "requestId";
+
+    @Mock
+    Defendant defendant;
+
+    @Mock
+    ProsecutionCase prosecutionCase;
+
+    @Mock
+    List<Offence> offences;
+
+    @Mock
+    private ProsecutionService prosecutionService;
+
+    @InjectMocks
+    private ProsecutionHandler prosecutionHandler;
+
+    @Test
+    @DisplayName("Tests successful get of prosecution document")
+    void testGetProsecutionDocument() throws ProsecutionServiceException, HandlerException {
+        when(prosecutionService.getDefendant(RESOURCE_URI)).thenReturn(defendant);
+        when(prosecutionService.getProsecutionCase("/" + defendant.getLinks().get("prosecution-case"))).thenReturn(prosecutionCase);
+        when(prosecutionService.getOffences(defendant.getLinks().get("offences"))).thenReturn(offences);
```

Comment:
And similar here, we can't be sure we are calling the prosecution service with the right link value

Reply:
Done

Reply:
Should we be using the specific value then?
```suggestion
        when(prosecutionService.getOffences(OFFENCES)).thenReturn(offences);
```

Reply:
This doesn't really mimic the functionality of the code. The purpose of the code is to retrieve the defendant resource and access the prosecution case / offences resource via retrieving the links of the defendant. Therefore, simply using `getOffences(OFFENCES)` does not really mimic the functionality of the code.

Reply:
Well we are using a mock object to find the expected parameter for the call to the service so I'd say we are not really testing the code. The value will be whatever we stub it to be and furthermore, by doing it this way we are making a call to a mock object that might have never happened otherwise (if the code didn't do it) preventing an unnecessary stubbing error. 

It seems we have now reverted back to how it was? If you changed the code to use a different key, I believe the test would still pass when it shouldn't.

Reply:
Sorry I see this is totally different now and we use real objects. It should make no difference then using `defendant.getLinks().get("offences")` or `OFFENCES` other than the latter being a bit more efficient.